### PR TITLE
Make label clause to be optionally

### LINKED ISSUE #839: We don't need LABEL while training a clustering mode
Related to #838 

```SQL
SELECT *
FROM iris.train
TRAIN sqlflow_models.DeepEmbeddingClusterModel
WITH
    model.n_clusters = 5,
    model.run_pretrain = false,
    model.pretrain_dims = [4,500,500,2000,10]
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class   -- useless
INTO sqlflow_models.clustering_model;
```

`LABEL` here in the `TRAIN` SQL is unnecessary.

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #839

### CODE REVIEW

**Review by typhoonzero — APPROVED** [review]

File: sql/codegen.go [hunk 1]
```diff
@@ -211,7 +211,19 @@ func newFiller(pr *extendedSelect, ds *trainAndValDataset, fts fieldTypes, db *D
...
 	// Default use int32 label dtype
 	labelDtype := "int32"
-	if dbDType, ok := fts.get(pr.label); ok {
+	var labelColumnName string
+	var e error
+
+	if pr.train {
+		labelColumnName = pr.label
+	} else {
+		r.TableName, labelColumnName, e = parseTableColumn(pr.into)
```
Comment (weiguoz) [inline sql/codegen.go:220]: How about merging with L215. Like:
```go
if r.TableName, labelColumnName, e := parseTableColumn(pr.into); e != nil {
    // ...
}
```

File: sql/python/sqlflow_submitter/db.py [hunk 2]
```diff
@@ -180,7 +180,9 @@ def reader():
             if driver == "mysql":
                 conn.ping(True)
             for row in rows:
-                label = row[label_idx] if label_idx is not None else None
+                # NOTE: If there is no label clause in the extened SQL, the default label value would 
+                # be -1, the Model implementation can determine use it or not.
+                label = row[label_idx] if label_idx is not None else -1
```
Comment (weiguoz) [inline sql/python/sqlflow_submitter/db.py:185]: I am worried that these two situations cannot be distinguished: 1) `row[label_idx] = -1`; 2) `else -1`

Comment (Yancey0623) [reply sql/python/sqlflow_submitter/db.py:185]: You're right, `-1` can not be used to distinguish `with label clause` and `without label clause`, the model implementation should deal with that.

File: sql/codegen.go (hunk 1 above)
Comment (Yancey0623) [reply sql/codegen.go:220]: Will polish it in the next PR.

**Pull request merged**